com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelUtil.java

```diff
@@ -0,0 +1,854 @@
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.IntBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Vector;
+import java.util.Map.Entry;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.ArrayUtils;
+
+import javax.vecmath.Matrix4d;
+import javax.vecmath.Matrix4f;
+import javax.vecmath.Point3d;
+import javax.vecmath.Point3f;
+import javax.vecmath.Quat4d;
+import javax.vecmath.Tuple3d;
+import javax.vecmath.Tuple4d;
+import javax.vecmath.Vector3d;
+import javax.vecmath.Vector3f;
+import javax.vecmath.Vector4d;
+
+import com.dynamo.bob.util.MathUtil;
+
+import com.dynamo.bob.util.MurmurHash;
+import com.dynamo.bob.util.RigUtil;
+import com.dynamo.bob.util.RigUtil.AnimationKey;
+import com.dynamo.bob.util.RigUtil.Weight;
+import com.dynamo.proto.DdfMath.Point3;
+import com.dynamo.proto.DdfMath.Quat;
+import com.dynamo.proto.DdfMath.Vector3;
+import com.dynamo.proto.DdfMath.Matrix4;
+import com.dynamo.proto.DdfMath.Transform;
+
+import com.dynamo.bob.pipeline.ModelImporter.Bone;
+import com.dynamo.bob.pipeline.ModelImporter.Mesh;
+import com.dynamo.bob.pipeline.ModelImporter.Model;
+import com.dynamo.bob.pipeline.ModelImporter.Node;
+import com.dynamo.bob.pipeline.ModelImporter.Options;
+import com.dynamo.bob.pipeline.ModelImporter.Scene;
+
+
+import com.dynamo.rig.proto.Rig;
+import com.dynamo.rig.proto.Rig.AnimationInstanceDesc;
+import com.dynamo.rig.proto.Rig.AnimationSetDesc;
+import com.google.protobuf.ByteString;
+
+public class ModelUtil {
+
+    public static Scene loadScene(byte[] content, String path, Options options) {
+        if (options == null)
+            options = new Options();
+        return ModelImporter.LoadFromBuffer(options, path, content);
+    }
+
+    public static Scene loadScene(InputStream stream, String path, Options options) throws IOException {
+        byte[] bytes = IOUtils.toByteArray(stream);
+        return loadScene(bytes, path, options);
+    }
+
+    public static void unloadScene(Scene scene) {
+    }
+
+    private static Transform toDDFTransform(ModelImporter.Transform transform) {
+        Vector3d translation = new Vector3d(transform.translation.x, transform.translation.y, transform.translation.z);
+        Vector3d scale = new Vector3d(transform.scale.x, transform.scale.y, transform.scale.z);
+        Quat4d rotation = new Quat4d(transform.rotation.x, transform.rotation.y, transform.rotation.z, transform.rotation.w);
+        return MathUtil.vecmathToDDF(translation, rotation, scale);
+    }
+
+    // private static Point3d toPoint3d(ModelImporter.Vec4 v) {
+    //     return new Point3d(v.x, v.y, v.z);
+    // }
```

Comment:
I'll keep these a little bit longer.
I should probably move them e.g. to RigUtil.java or MathUtil.java or similar